hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3GuardTtl.java

```diff
@@ -356,4 +358,14 @@ public void testListingFilteredExpiredItems() throws Exception {
     }
   }
 
+  protected DirListingMetadata getDirListingMetadata(final MetadataStore ms,
```

Comment:
I don't see how this change is connected to the original issue.

Reply:
I was getting an NPE during test runs where a list op was returning null but the tests assumed it never did. Added for debugging

Reply:
correction. You were reporting an NPE on test runs which I wasn't seeing. This change is to debug your test failures